qmetaobject/src/qtdeclarative.rs

```diff
@@ -436,6 +436,79 @@ pub fn qml_register_type<T: QObject + Default + Sized>(
     }
 }
 
+cpp! {{
+    typedef QObject *(*QmlRegisterSingletonTypeCallback)(QQmlEngine *, QJSEngine *);
+}}
+
+/// Register a default-constructed object of specified type as a singleton QML object.
+///
+/// A new object will be created for each new instance of `QmlEngine`.
+///
+/// Refer to the Qt documentation for qmlRegisterSingletonType.
+pub fn qml_register_singleton_type<T: QObject + Sized + Default>(
+    uri: &std::ffi::CStr,
+    version_major: u32,
+    version_minor: u32,
+    type_name: &std::ffi::CStr,
+) {
+    let uri_ptr = uri.as_ptr();
+    let type_name_ptr = type_name.as_ptr();
+
+    extern "C" fn callback_fn<T: QObject + Default + Sized>(
+        _qml_engine: *mut c_void,
+        _js_engine: *mut c_void,
+    ) -> *mut c_void {
+        let obj: Box<RefCell<T>> = Box::new(RefCell::new(T::default()));
+        let obj_ptr = unsafe { T::cpp_construct(&obj) };
+        Box::into_raw(obj);
+        obj_ptr
+    };
+    let callback_fn: extern "C" fn(
+        _qml_engine: *mut c_void,
+        _js_engine: *mut c_void,
+    ) -> *mut c_void = callback_fn::<T>;
+
+    unsafe {
+        cpp!([uri_ptr as "const char*", version_major as "int", version_minor as "int",
+                type_name_ptr as "const char*", callback_fn as "QmlRegisterSingletonTypeCallback"] {
+            qmlRegisterSingletonType<QObject>(uri_ptr, version_major, version_minor, type_name_ptr,
+                callback_fn);
+        })
+    }
+}
+
+/// Register the passed object as a singleton QML object.
+///
+/// The object is shared between all instances of `QmlEngine`.
+///
+/// Refer to the Qt documentation for qmlRegisterSingletonInstance.
+/// Only avaiable in Qt 5.14 or above.
+#[cfg(qt_5_14)]
+pub fn qml_register_singleton_instance<T: QObject + Sized + Default>(
+    uri: &std::ffi::CStr,
+    version_major: u32,
+    version_minor: u32,
+    type_name: &std::ffi::CStr,
+    obj: T,
+) {
+    let uri_ptr = uri.as_ptr();
+    let type_name_ptr = type_name.as_ptr();
+
+    let obj_box = Box::new(RefCell::new(obj));
+    let obj_ptr = unsafe { T::cpp_construct(&obj_box) };
+    Box::into_raw(obj_box);
+
+    unsafe {
+        cpp!([uri_ptr as "char*", version_major as "int", version_minor as "int",
+                type_name_ptr as "char*", obj_ptr as "QObject*"] {
+            #ifdef QT_5_14
```

Comment:
You do not need this macro: you can do 

```suggestion
            #if QT_VERSION >= QT_VERSION_CHECK(5,14,0)
```

Reply:
Changed.